system/core/HyphaSession.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+	/**
+	 * HyphaSession manages the PHP session for the current request.
+	 * It should always be used instead of direct access to
+	 * $_SESSION.
+	 */
+	class HyphaSession {
```

Comment:
To ensure the lock it would be better to apply the singleton pattern;


```
class HyphaSession {
    private static $instance;
    public static function getInstance() {
        if (!self::$instance instanceof HyphaSession) {
            self::$instance = new HyphaSession;
        }

        return self::$instance;
    }

    private function __construct() {
        // ....
    }
}

$hyphaSession = HyphaSession::getInstance();
```

Reply:
Done.